diesel_codegen_shared/src/schema_inference/pg.rs

```diff
@@ -99,14 +110,12 @@ pub fn get_primary_keys(conn: &PgConnection, table_name: &str) -> QueryResult<Ve
 }
 
 pub fn load_table_names(connection: &PgConnection) -> Result<Vec<String>, Box<Error>> {
-    use diesel::expression::dsl::sql;
-
-    let query = select(sql::<types::VarChar>("table_name FROM information_schema.tables"))
-        .filter(sql::<types::Bool>("\
-            table_schema = 'public' AND \
-            table_name NOT LIKE '\\_\\_%' AND \
-            table_type LIKE 'BASE TABLE'\
-        "));
+    use self::information_schema::tables::dsl::*;
+
+    let query = tables.select(table_name)
+        .filter(table_schema.eq("public"))
+        .filter(table_name.not_like("\\_\\_%"))
```

Comment:
The `_` character in a SQL `LIKE` expression is equivalent to `.` in a regular expression. `\\_` is matching a literal `_` character in the resulting string.